File: Data Structures For Interview/JavaScript/Matrix Programs/Sum.js
function sum(m1){

    let row=[];
    let col=[];

    let rowSum=0;
    let colSum=0;

    for(let i=0;i<m1.length;i++){
        rowSum=0;
        for(let j=0;j<m1[0].length;j++){
            rowSum+=m1[i][j];
        }
        row[i]=rowSum;
    }
    
    for(let i=0;i<m1.length;i++){
        colSum=0;
        for(let j=0;j<m1[0].length;j++){
            colSum+=m1[j][i];
        }
        col[i]=colSum;
    }

    for(let i=0;i<row.length;i++){
        console.log("Sum of "+(i+1)+" th row is :- "+row[i]);
    }
    for(let i=0;i<col.length;i++){
        console.log("Sum of "+(i+1)+" th column is :- "+col[i]);
    }
}

let m1=[
    [1,2,3],
    [4,5,6],
    [7,8,9]
];

let m2=[
    [1,2,3],
    [4,5,6],
    [7,8,9]
];

sum(m1);

File: Data Structures For Interview/Java/Matrix Programs/Sum.java
public class Sum {

    static void sum(int m[][]){

        int row=m.length;
        int col=m[0].length;

        int rowSum[]=new int[row];
        int colSum[]=new int[col];

        int rowItem=0;
        int colItem=0;

        for(int i=0;i<row;i++){
            rowItem=0;
            for(int j=0;j<col;j++){
                rowItem+=m[i][j];
            }
            rowSum[i]=rowItem;
        }

        for(int i=0;i<row;i++){
            colItem=0;
            for(int j=0;j<col;j++){
                colItem+=m[j][i];
            }
            colSum[i]=colItem;
        }

        for(int i=0;i<rowSum.length;i++){
            System.out.println("Sum of "+(i+1)+" th row  is :- "+rowSum[i]);
        }
        for(int i=0;i<colSum.length;i++){
            System.out.println("Sum of "+(i+1)+" th column is :- "+colSum[i]);
        }
    }
    public static void main(String[] args) {

        int m[][]={
            {1,2,3},
            {4,5,6},
            {7,8,9}
        };
        sum(m);
    }
}
